tests/integration/api/test_activation.py

```diff
@@ -252,24 +252,76 @@ def test_create_activation_bad_entity(client: APIClient):
 
 @pytest.mark.django_db(transaction=True)
 def test_create_activation_unprocessible_entity(client: APIClient):
-    test_activation = {
-        "name": "test-activation",
-        "description": "test activation",
-        "is_enabled": True,
-        "decision_environment_id": 32,
-        "rulebook_id": 100,
-    }
+    fks = create_activation_related_data()
+    test_activation = TEST_ACTIVATION.copy()
+    test_activation["decision_environment_id"] = fks["decision_environment_id"]
+    test_activation["project_id"] = fks["project_id"]
+    test_activation["rulebook_id"] = fks["rulebook_id"]
+    test_activation["extra_var_id"] = fks["extra_var_id"]
 
+    # test with non-existent decision environment
+    with mock.patch.object(
+        serializers.ActivationCreateSerializer,
+        "create",
+        mock.Mock(side_effect=IntegrityError),
+    ):
+        response = client.post(
+            f"{api_url_v1}/activations/",
+            data={
+                **test_activation,
+                "decision_environment_id": 0,
+            },
+        )
+    assert response.status_code == status.HTTP_422_UNPROCESSABLE_ENTITY
+    assert (
+        response.data["detail"]
+        == "Decision environment with ID=0 does not exist."
+    )
+    # test with non-existent rulebook
+    with mock.patch.object(
+        serializers.ActivationCreateSerializer,
+        "create",
+        mock.Mock(side_effect=IntegrityError),
```

Comment:
This test is only for testing whether a meaningful response is created on error. Shouldn't you add a real test for non-existing rulebook id without a mock?